src/mscorlib/shared/System/ReadOnlyMemory.cs

```diff
@@ -159,25 +195,37 @@ public ReadOnlySpan<T> Span
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get
             {
-                if (_index < 0)
-                    return ((OwnedMemory<T>)_arrayOrOwnedMemory).Span.Slice(_index & RemoveOwnedFlagBitMask, _length);
-                return new ReadOnlySpan<T>((T[])_arrayOrOwnedMemory, _index, _length);
+                return
+                    _index < 0 ? ((OwnedMemory<T>)_object).Span.Slice(_index & RemoveOwnedFlagBitMask, _length) :
```

Comment:
nit: this is difficult to parse. can u split the conditions into if blocks?

Reply:
ok